docs/modules/ROOT/pages/how-to-guides/proc_upgrade_build_pipeline.adoc

```diff
@@ -58,3 +58,20 @@ After upgrading the build pipeline, you can also customize it:
 {ProductName} creates pull requests that contain updates to the pipeline definitions under the `.tekton` directory.
 Run the tests on these pull requests by creating a comment containing `/ok-to-test` and then merging the changes when the tests pass.
 ====
+
+.Security
+To further reinforce the security of your custom build pipeline, complete the following steps:
+
+* Add an `OWNERS` file to the root of your repository and list trusted contributors there.
+** To learn more, see Kubernetes docs about link:https://www.kubernetes.dev/docs/guide/owners/[OWNERS files].
+* Avoid commenting `/ok-to-test` on pull requests from untrusted authors, review such PRs carefully first. The `/ok-to-test` comment runs the PipelineRun, and malicious code in a PR can change your build and compromise the security of your application.
+** To learn more about running the PipelineRun, see the link:https://pipelinesascode.com/docs/guide/running/#running-the-pipelinerun[Running the PipelineRun guide] from Pipelines as Code docs. 
+* Consider changing the PipelineRun definition source by setting the `pipelinerun_provenance` setting to `default_branch`. With this setting, Pipelines as Code uses the PipelineRun definition from the default branch of the repository, usually `main` or `master`, and only contributors with default branch merge rights can modify the PipelineRun.
++
+If you don’t set `pipelinerun_provenance`, you allow the default behaviour: the PipelineRun definition is fetched from the branch where the PipelineRun event is triggered, and any contributor can get access to the PipelineRun definition.
```

Comment:
```suggestion
If you don’t set `pipelinerun_provenance`, you allow the default behaviour: the PipelineRun definition is fetched from the branch where the PipelineRun event is triggered, and any contributors can get access to the PipelineRun definition.
```

you may want to define what is the meaning of a contributor in pac terms, it means someone who is a contributor on a repository and any members of the organisation where the repository belongs.

Reply:
@chmouel Thanks for being attentive to details, appreciate it. 
There's also a longer list of who can run pipelinerun on CI: https://pipelinesascode.com/docs/guide/running/ (in the beginning). In our case does this whole list apply? If yes, do you think we can just link to this list? To avoid rewriting a long piece of text that already exists in other docs.

Reply:
@missmesss yes, all those rules would applies on appstudios you can link there!